### ISSUE TITLE
Pause/Resume Training - Language Model from Scratch

### ISSUE THREAD

**Issue by P1983** [ThilinaRajapakse/simpletransformers#684]
Hi Thilina, first of all, I'd like to thank you for this wonderfull job. Simpletransformers is a great library!

As I'm working with limited and shared resources, I'd like to know **if it's possible to pause and resume a language model train from scratch.** I mean: **If I press Ctrl - C and stop my training, can I start again, from the last checkpoint?**

I saw one related question of a user who wanted to pause a classification model, who should point to it's last checkpoint to continue training. But as we should inform 'None' for training a model from Scratch, wich I presume, may have some training particularities, I'd like to know for shure, if it's possible to pause a training and it's correct way to start again.

**In addition, I'd like to check if I'm facing a bug or not in some of the model args.** I know it's not recommended to put different things in the same issue, but in this case, I think they are strongly related, since I believe the model needs it's checkpoint to continue. Thanks in advance for all your attention!

**To Reproduce**
Using:
I'm using the following args:
model_args = LanguageModelingArgs()
model_args.reprocess_input_data = False
model_args.overwrite_output_dir = True
model_args.num_train_epochs = 4
model_args.dataset_type = "simple"
model_args.do_lower_case = True
model_args.encoding = "utf-8"
model_args.max_seq_length = 128
model_args.train_batch_size = 8
model_args.eval_batch_size = 8
model_args.learning_rate = 2e-5
model_args.vocab_size = 100000
model_args.min_frequency = 2
model_args.output_dir = "outputs/LM/PreTrain"
**model_args.no_cache = True
model_args.evaluate_during_training = True
model_args.evaluate_during_training_verbose = True
model_args.save_eval_checkpoints = False
model_args.save_model_every_epoch = False**
model_args.strip_accents = True
model_args.clean_text = True
model_args.warmup_steps = 1000
model_args.gradient_accumulation_steps = 1
model_args.block_size = 128

**Expected behavior**
I expected not keep model checkpoint in the output_dir, but they're all being saved for each 2000 steps. And I also though that the eval_results would be printed in the jupyter log output, but they aren't. I can only check it's best version in the output_dir.

**Screenshots**

**Output_dir with checkpoint sample:**
drwxrwxr-x 2 f8155859 f8155859  215 Aug 28 14:42 checkpoint-36000
drwxrwxr-x 2 f8155859 f8155859  215 Aug 28 17:41 checkpoint-38000
drwxrwxr-x 2 f8155859 f8155859  215 Aug 28 20:28 checkpoint-40000
drwxrwxr-x 2 f8155859 f8155859  215 Aug 28 23:17 checkpoint-42000
drwxrwxr-x 2 f8155859 f8155859  215 Aug 29 02:05 checkpoint-44000
drwxrwxr-x 2 f8155859 f8155859  215 Aug 29 04:53 checkpoint-46000

**Jupyter log for training method**
HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Selected optimization level O1:  Insert automatic casts around Pytorch functions and Tensor methods.

Defaults for this optimization level are:
enabled                : True
opt_level              : O1
cast_model_type        : None
patch_torch_functions  : True
keep_batchnorm_fp32    : None
master_weights         : None
loss_scale             : dynamic
Processing user overrides (additional kwargs that are not None)...
After processing overrides, optimization options are:
enabled                : True
opt_level              : O1
cast_model_type        : None
patch_torch_functions  : True
keep_batchnorm_fp32    : None
master_weights         : None
loss_scale             : dynamic

HBox(children=(FloatProgress(value=0.0, description='Epoch', max=4.0, style=ProgressStyle(description_width='i…

HBox(children=(FloatProgress(value=0.0, description='Running Epoch 0 of 4', max=158321.0, style=ProgressStyle(…

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0
Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 32768.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 65536.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

HBox(children=(FloatProgress(value=0.0, max=1266566.0), HTML(value='')))

Gradient overflow.  Skipping step, loss scaler 0 reducing loss scale to 131072.0

HBox(children=(FloatProgress(value=0.0, max=9917241.0), HTML(value='')))

**Desktop (please complete the following information):**
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.43       Driver Version: 440.43       CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GRID V100DX-16C     On   | 00000000:02:01.0 Off |                    0 |
| N/A   N/A    P0    N/A /  N/A |   9143MiB / 16320MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

**Comment by ThilinaRajapakse** [comment]
You should be able to resume training from a saved checkpoint. Setting `model_name` to `None` essentially makes sure that the model weights are randomly initialized, but that's not something we want to do when resuming from a checkpoint!

To see the evaluation results in the log, you can set `evaluate_during_training_verbose` to `True` in the model_args.

Simple Transformers will checkpoint all models at every 2000 steps by default. This can be changed by adjusting `save_steps` (and turned off by setting it to -1).

**Comment by P1983** [comment]
Thank you for the reply. I have made the adjustments and it works fine.
